recipes/libhal/all/conanfile.py

```diff
@@ -74,18 +86,27 @@ def build(self):
     def package(self):
         copy(self, "LICENSE", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
         copy(
-            self, 
-            "*.h", 
-            dst=os.path.join(self.package_folder, "include"), 
+            self,
+            "*.h",
+            dst=os.path.join(self.package_folder, "include"),
             src=os.path.join(self.source_folder, "include")
         )
         copy(
-            self, 
-            "*.hpp", 
-            dst=os.path.join(self.package_folder, "include"), 
+            self,
+            "*.hpp",
+            dst=os.path.join(self.package_folder, "include"),
             src=os.path.join(self.source_folder, "include")
         )
 
     def package_info(self):
         self.cpp_info.bindirs = []
+        self.cpp_info.frameworkdirs = []
         self.cpp_info.libdirs = []
+        self.cpp_info.resdirs = []
+
+        version = Version(self.version)
+        if self._bare_metal and version < "3.0.0":
+            self.cpp_info.defines = [
+                "BOOST_LEAF_EMBEDDED",
+                "BOOST_LEAF_NO_THREADS"
```

Comment:
Are these defines really provided by libhal, or actually Boost should export it in case of using baremetal?

Reply:
These are provided by libhal for versions below 3.0.0. Not Boost.

Reply:
Also, Boost could emit the BOOST_LEAF_EMBEDDED if the OS is baremetal, but the NO_THREADS flag is one of 3 options. Another one is FREERTOS and another allows a hook for TLS support for RTOSs not currently supported directly. So this would have to be another option for the boost library. Note that I plan to drop Boost.LEAF and boost after 3.0.0 in favor of C++ exceptions, so there won't be new dependencies on Boost.LEAF in the future.